src/engine/bufferscalers/enginebufferscalerubberband.cpp

```diff
@@ -286,3 +281,30 @@ int EngineBufferScaleRubberBand::runningEngineVersion() {
     return 2;
 #endif
 }
+
+void EngineBufferScaleRubberBand::reset() {
+    m_pRubberBand->reset();
+
+    // As mentioned in the docs (https://breakfastquay.com/rubberband/code-doc/)
+    // and FAQ (https://breakfastquay.com/rubberband/integration.html#faqs), you
+    // need to run some silent samples through the time stretching engine first
+    // before using it. Otherwise it will eat add a short fade-in, destroying
+    // the initial transient.
+    size_t remaining_padding = m_pRubberBand->getPreferredStartPad();
```

Comment:
We still support Rubberband V2 so you'll need to add the appropriate ifdefs for it to compile on older versions (use the `RUBBERBANDV3` `#define`).

Reply:
No problem, I had to approve the CI run first since you are technically still a first-time-contributor ;)